### PR TITLE
FEATURE: add swipe up to close toast notification

### PR DESCRIPTION
**Pull request by dbattersby**

### CODE REVIEW

File: app/assets/javascripts/float-kit/addon/components/d-default-toast.gjs [hunk 1]
```diff
@@ -1,63 +1,155 @@
...
+      enableBodyScroll(this.wrapperElement);
+    } else {
+      disableBodyScroll(this.wrapperElement);
+    }
+
+    this.scrollLocked = !this.scrollLocked;
+  }
+
+  @action
+  cleanupToast() {
```
Comment (jjaffeux) [inline app/assets/javascripts/float-kit/addon/components/d-default-toast.gjs:85]: when is this called?

File: app/assets/javascripts/float-kit/addon/components/d-default-toast.gjs [hunk 2]
```diff
@@ -1,63 +1,155 @@
...
+  }
+
+  <template>
+    <div
+      class={{concatClass
+        "fk-d-default-toast"
+        (concat "-" (or @data.theme "default"))
+      }}
+      ...attributes
+      {{didInsert this.setupToast}}
```
Comment (jjaffeux) [inline app/assets/javascripts/float-kit/addon/components/d-default-toast.gjs:99]: for cleanupToast I suggest you make setupToast a modifier so you can have didInsert/willDestroy in one function:


```
setupToast = modifier(... => {
  addListener(...)
  
  return () => {
    removeLsitener(...);
  };
});

Comment (dbattersby) [reply app/assets/javascripts/float-kit/addon/components/d-default-toast.gjs:99]: I forgot to add the willDestroy for the cleanup, but I can use the modifier approach to handle this instead 👍

File: app/assets/javascripts/float-kit/addon/components/d-default-toast.gjs [hunk 3]
```diff
@@ -1,63 +1,154 @@
...
+
+    await this.wrapperElement.animate([{ transform: `translateY(-150px)` }], {
+      duration: 500,
+      fill: "forwards",
+    });
+
+    this.animating = false;
+  }
+
+  toggleLock() {
```
Comment (jjaffeux) [inline app/assets/javascripts/float-kit/addon/components/d-default-toast.gjs:77]: you can just use the @bind decorator on this, instead of the .bind(...)

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**